src/TufValidatedComposerRepository.php

```diff
@@ -173,12 +173,18 @@ protected function configurePackageTransportOptions(PackageInterface $package):
     /**
      * Indicates if TUF is enabled for this repository.
      *
+     * @param \Composer\Package\PackageInterface|null $package
+     *   A specific package being validated, if any.
+     *
      * @return bool
      *   Whether PHP-TUF is enabled for this repository.
      */
-    private function isTufEnabled(): bool
+    private function isTufEnabled(?PackageInterface $package = null): bool
     {
-        return $this->updater instanceof ComposerCompatibleUpdater;
+        // Metapackages are not downloaded into the code base at all, so don't
+        // bother validating them.
+        // @see https://github.com/composer/composer/blob/11e5237ad9d9e8f29bdc57d946f87c816320d863/doc/07-runtime.md?plain=1#L110
+        return $this->updater instanceof ComposerCompatibleUpdater && $package?->getType() !== 'metapackage';
```

Comment:
So the meta package must at least have composer.json, or something else to specify what packages it needs. Is that downloaded to inspect it?

Reply:
Well, all Composer packages must have a composer.json, metapackage or no.

And yes, it's getting downloaded -- I updated the tests to confirm that. As you can see, we still install Pathauto and Token, but we install them _via the metapackage_ rather than individually. That pretty much proves it works.